sunpy/net/fido_factory.py

```diff
@@ -162,6 +173,16 @@ def file_num(self):
         """
         return self._numfile
 
+    @property
```

Comment:
So a UnifiedResponse can be sliced by name or number so I think the errors would need to work in a similar way and also update the errors accordingly? So given a response from two clients 

```
res.errors # would return all errors/None
res[0].errors # would only return error from first client
res['a'].errors # would only return error from 'a' client
``` 

or 

```
res.errors[0] # would only return error from first client
res.errors['a'] # would only return error from 'a' client
```

I think the first option would only require the `__getitem__` to be updated and is maybe better anyway @Cadair what do you think?

Reply:
As Shane says, we should update `__getitem__`